src/opm/output/eclipse/Summary.cpp

```diff
@@ -1577,6 +1577,73 @@ quantity well_guiderate(const fn_args& args)
     return guiderate_value<i>(xwPos->second.guide_rates);
 }
 
+quantity well_efficiency_factor(const fn_args& args)
+{
+    const auto zero = quantity { 0.0, measure::identity };
+
+    if (args.schedule_wells.empty()) {
+        return zero;
+    }
+
+    const auto* well = args.schedule_wells.front();
```

Comment:
Is there only one relevant well?

Reply:
> Is there only one relevant well?

In this case, yes.  The `well_efficiency_factor()` implements summary vector `WEFF` which applies to single wells.  This function will be called once for each well explicitly named in the `WEFF` summary keyword, e.g.,
```
WEFF
  'P-1' 'P-2' 'P-3' 'I-10'
/
```
 or once for each well in the model if there are no explicitly named wells, e.g.,
```
WEFF
/
```
We use the same construction in [`well_efficiency_factor_grouptree()`](https://github.com/OPM/opm-common/blob/851f8e063272b99afc513c54acd0e825e78144d5/src/opm/output/eclipse/Summary.cpp#L1601) which implements the underlying logic to calculate the `WEFFG` summary vector for one well at a time.

Reply:
thanks.